lib/ii.c

```diff
@@ -11766,21 +11766,31 @@ grn_ii_select_data_check_near_element_intervals(grn_ctx *ctx,
     }
     return true;
   } else if (data->mode == GRN_OP_ORDERED_NEAR_PHRASE) {
-    uint32_t i;
-    uint32_t n = data->n_token_infos;
-    if (n > n_max_element_intervals + 1) {
-      n = n_max_element_intervals + 1;
-    }
+    uint32_t i_token_info;
+    uint32_t n_token_infos = data->n_token_infos;
+    uint32_t i_phrase;
+    uint32_t n_phrases = data->n_phrases;
+    if (n_phrases > n_max_element_intervals + 1) {
+      n_phrases = n_max_element_intervals;
+    }
```

Comment:
Ah, we should improve variable name. How about this?

```suggestion
    uint32_t i_interval;
    uint32_t n_intervals = data->n_phrases - 1;
    if (n_intervals > n_max_element_intervals) {
      n_intervals = n_max_element_intervals;
    }
```

Reply:
I adopted your suggestion.

Reply:
Could you push it?

Reply:
Sorry, I pushed it.